### PR TITLE
Add FAQ entries for parameters in queries and SpEL.

### PR DESCRIPTION
**Pull request by michael-simons**

### CODE REVIEW

File: docs/faq/faq.adoc [hunk 1]
```diff
@@ -134,6 +134,43 @@ And here is the reactive version, omitting the setup for brevity:
...
+You do this exactly the same way as in a standard Cypher query issued in the Neo4j Browser or the Cypher-Shell,
+with the `$` syntax (from Neo4j 4.0 on upwards, the old `{foo}` syntax for Cypher parameters has been removed from the database);
+
+[source,java,indent=0]
+[[custom-queries-with-parameters]]
+.ARepository.java
+----
+include::../../examples/docs/src/test/java/org/neo4j/doc/springframework/data/docs/repositories/domain_events/ARepository.java[tags=standard-parameter]
+----
+<.> Here we are referring to the parameter by it's name.
```
Comment (meistermeier) [inline docs/faq/faq.adoc]:
```suggestion
<.> Here we are referring to the parameter by its name.
```

File: docs/faq/faq.adoc [hunk 2]
```diff
@@ -134,6 +134,43 @@ And here is the reactive version, omitting the setup for brevity:
...
+    You can also use `$0` etc. instead.
+
+NOTE: You need to compile your Java 8+ project with `-parameters` to make named parameters
+      work without further annotations. The Spring Boot Maven and Gradle plugins do this
+      automatically for you. If this is not feasible for any reason, you can either add
+      `@Param`  and specify the name explicitly or used the parameters index.
```
Comment (meistermeier) [inline docs/faq/faq.adoc]:
```suggestion
      `@Param`  and specify the name explicitly or use the parameters index.
```

File: docs/faq/faq.adoc [hunk 3]
```diff
@@ -134,6 +134,43 @@ And here is the reactive version, omitting the setup for brevity:
...
+The following example basically defines the same query as above, but uses a `WHERE` clause to avoid even more curly braces:
+
+[source,java,indent=0]
+[[custom-queries-with-spel]]
+.ARepository.java
+----
+include::../../examples/docs/src/test/java/org/neo4j/doc/springframework/data/docs/repositories/domain_events/ARepository.java[tags=spel]
+----
+
+The SpEL blocked starts with `:#{` and than refers to the given `String`  parameters by name (`#pt1`).
```
Comment (meistermeier) [inline docs/faq/faq.adoc]:
```suggestion
The SpEL blocked starts with `:#{` and than refers to the given `String` parameters by name (`#pt1`).
```

Comment (michael-simons) [reply docs/faq/faq.adoc]: I am not doing this on purpose, I swear!

**Pull request merged**